src/ln/peer_handler.rs

```diff
@@ -37,6 +37,9 @@ pub trait SocketDescriptor : cmp::Eq + hash::Hash + Clone {
 	/// indicating that read events on this descriptor should resume. A resume_read of false does
 	/// *not* imply that further read events should be paused.
 	fn send_data(&mut self, data: &Vec<u8>, write_offset: usize, resume_read: bool) -> usize;
+	///Indicate to the network provider that we want to disconnect this peer, no more allowing any call
+	///to disconnect_event or read_event
+	fn notify_disconnection(&mut self);
```

Comment:
Maybe call this "disconnect_socket" instead, and the comment needs flushed out more since its an external API that's easy to mess up:
```
/// Disconnect the socket pointed to by this SocketDescriptor. Once this function returns, no
/// more calls to write_event, read_event or disconnect_event may be made with this descriptor.
/// No disconnect_event should be generated as a result of this call, though obviously races
/// may occur whereby disconnect_socket is called after a call to disconnect_event but prior to
/// that event completing.
```
should be sufficient, no?